examples/threading/threading.c

```diff
@@ -13,8 +13,29 @@ void* threadfunc(void* thread_param)
 
     // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
     // hint: use a cast like the one below to obtain thread arguments from your parameter
-    //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
-    return thread_param;
+    struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    int ret;								 //to check the return value for different operations
+
+    ret = usleep((thread_func_args->wait_to_obtain_ms)*1000);			 //wait for required msec to obtain the mutex
+    if(ret==0){thread_func_args->thread_complete_success = true;}		 //check conditions if usleep succeeded
```

Comment:
@Mich2899 is the thread complete with success here?   What would happen if another thread was checking the variable at this point?